src/mcpdotnet/Protocol/Transport/SseClientTransport.cs

```diff
@@ -110,7 +110,7 @@ public override async Task SendMessageAsync(
             throw new InvalidOperationException("Transport not connected");
 
         using var content = new StringContent(
-            JsonSerializer.Serialize(message, _jsonOptions),
+            JsonSerializer.Serialize(message, _jsonOptions.GetTypeInfo<IJsonRpcMessage>()),
```

Comment:
Is this going to correctly serialize implementations of IJsonRpcMessage, including public properties on those implementations?